client/components/purchase-detail/index.jsx

```diff
@@ -37,16 +38,34 @@ const PurchaseDetail = ( {
 		);
 	}
 
+	let requiredNotice;
+
+	if ( requiredText ) {
+		requiredNotice = (
+			<div className="purchase-detail__required-notice">
+				<em>{ requiredText }</em>
+			</div>
+		);
+	}
+
 	return (
 		<div className={ classes }>
-			{ icon && <div className="purchase-detail__icon"><Gridicon icon={ icon } /></div> }
+			{ requiredNotice }
```

Comment:
I already mentioned it in https://github.com/Automattic/wp-calypso/pull/3715#discussion_r54707519 but since the condition to trigger the notice is very simple we could inline everything:

```javascript
{ requiredText &&
  <div className="purchase-detail__required-notice">
     <em>{ requiredText }</em>
  </div>
}
```